src/assets/specs/command-api.yaml

```diff
@@ -0,0 +1,108 @@
+openapi: "3.0.0"
+info:
+  title: "Rancher Desktop API"
+  version: 0.0.1
+paths:
+  /v0/list-settings:
+    get:
+      operationId: listSettings
+      summary:  List the current preference settings
+      responses:
+        '200':
+          description: |-
+            200 response
+          content:
+            application/json:
+              schema:
+                "$ref" : "#/components/schemas/preferences"
+
+  /v0/set:
+    put:
+      operationId: updateSettings
+      summary:  Updates the specified preference settings
+      requestBody:
+        description: |-
+          JSON block consisting of some or all of the current preferences,
+          with changes applied to any number of settings the backend supports changing this way.
+        content:
+          application/json:
+            schema:
+              "$ref" : "#/components/schemas/preferences"
+        required: true
+      responses:
+        '202':
+          description: |-
+            202 response
+          content:
+            "text/plain":
+              schema:
+                type: string
+        '400':
+          description: |-
+            400 response
+          content:
+            "text/plain":
+              schema:
+                type: string
+
+  /v0/shutdown:
+    put:
+      operationId: shutdownApp
+      summary:  Shuts down Rancher Desktop
+      responses:
+        '202':
+          description: |-
+            202 response
+          content:
+            "text/plain":
+              schema:
+                type: string
+components:
+  schemas:
+    preferences:
+      type: object
+      properties:
+        kubernetes:
+          type: object
+          properties:
+            version:
+              type: string
+            memoryInGB:
+              type: integer
+              minimum: 1
+            numberInCPUs:
+              type: integer
+              minimum: 1
+            port:
+              type: integer
+            containerEngine:
+              type: string
+            checkForExistingKimBuilder:
+              type: boolean
+            enabled:
+              type: boolean
+            WSLIntegrations:
+              type: object
```

Comment:
I _think_ there's some way to express a map of arbitrary keys and values of a given shape, but I don't know what it is.

Reply:
Found it:
```
WSLIntegrations:
              type: object
              additionalProperties:
                type: boolean
```
Not ideal, but kind of logical.